packages/ui/src/components/SelectInputV2/SearchBarDropdown.tsx

```diff
@@ -0,0 +1,181 @@
+import styled from '@emotion/styled'
+import { Icon } from '@ultraviolet/icons'
+import type { Dispatch, SetStateAction } from 'react'
+import { TextInputV2 } from '../TextInputV2'
+import type { DataType, OptionType } from './types'
+
+type SearchBarProps = {
+  options: DataType
+  onSearch: Dispatch<SetStateAction<DataType>>
+  placeholder: string
+  displayedOptions: DataType
+  multiselect: boolean
+  searchInput: string | undefined
+  setSearchInput: Dispatch<SetStateAction<string>>
+  selectedValues: OptionType[]
+  setSelectedValues: Dispatch<SetStateAction<OptionType[]>>
+  setSearchBarActive: Dispatch<SetStateAction<boolean>>
+  onChange?: (value: string[]) => void
+  selectAll?: boolean
+  setAllSelected: Dispatch<SetStateAction<boolean>>
+  selectAllGroup?: boolean
+  selectedGroups: string[]
+  setSelectedGroups: Dispatch<SetStateAction<string[]>>
+  numberOfOptions: number
+}
+
+const StyledInput = styled(TextInputV2)`
+  margin: ${({ theme }) => theme.space[2]};
+`
+
+const findClosestOption = (
+  options: DataType,
+  searchInput: string | undefined,
+) => {
+  if (searchInput) {
+    if (!Array.isArray(options)) {
+      const possibleOptions = { ...options }
+      Object.keys(possibleOptions).map((group: string) => {
+        possibleOptions[group] = possibleOptions[group].filter(
+          option => !option.disabled,
+        )
+
+        return null
+      })
+      if (
+        Object.keys(possibleOptions).some(
+          group => possibleOptions[group].length > 0,
+        )
+      ) {
+        const firstFit = Object.keys(possibleOptions)
+          .map(group => possibleOptions[group][0])
+          .filter(value => !!value)[0]
+
+        return firstFit
+      }
+    } else {
+      const possibleOptions = [...options].filter(option => !option.disabled)
+
+      if (possibleOptions.length > 0) {
+        return possibleOptions[0]
+      }
+    }
+  }
+
+  return 'NO_MATCH'
```

Comment:
is there a way to customize/localize text providing `locales` prop like some component in `ultraviolet/plus ?

Reply:
this text will never be shown to the user, it is just a way to know when there is no match and thus do nothing in that case, so I don't think having locales is important here?

Reply:
this text will never be shown to the user, it is just a way to know when there is no match and thus do nothing in that case, so I don't think having locales is important here?